src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/DefaultLSPDocumentSynchronizer.cs

```diff
@@ -243,12 +243,19 @@ private class DocumentSynchronizingContext
                     RequiredHostDocumentVersion = requiredHostDocumentVersion;
                     RejectOnNewerParallelRequest = rejectOnNewerParallelRequest;
                     _onSynchronizedSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
-                    _cts = CancellationTokenSource.CreateLinkedTokenSource(requestCancellationToken);
 
-                    // This cancellation token is the one passed in from the call-site that needs to synchronize an LSP document with a virtual document.
-                    // Meaning, if the outer token is cancelled we need to fail to synchronize.
-                    _cts.Token.Register(() => SetSynchronized(false));
-                    _cts.CancelAfter(timeout);
+                    _cts = CancellationTokenSource.CreateLinkedTokenSource(requestCancellationToken);
```

Comment:
Should we just check the requestCancellationToken for cancellation instead of adding to the linked source? That way it's clear what scenario this is handling

Reply:
I had it that way at first, but `SetSynchronized` uses `_cts` so I tried it the way that I needed to make the least changes.

True that the reason for this is unclear just from code, I'll add a comment.

Reply:
FWIW I had this same question at first, but did work out the answer (and hence didn't bother commenting) so its not the _worst_ thing in the world. A comment is still a good idea though 👍

Reply:
Yea, if I had an easier way to navigate through code in PRs I probably would have tried. Laziness be my name

Reply:
Yeah, "GoToImplementation" support in GitHub would be pretty nice.